x-pack/plugins/observability/server/services/slo/transform_generators/transform_generator.ts

```diff
@@ -70,7 +85,11 @@ export abstract class TransformGenerator {
                 },
               };
             },
-            { 'slo.instanceId': { terms: { field: slo.groupBy } } }
+            {
```

Comment:
Above, we need to change the conditions to create the groupings variable:

```
    const groupings = ![slo.groupBy]
      .flat()
      .filter((value) => !!value)
      .includes(ALL_VALUE)
      ? [slo.groupBy].flat().reduce(
          (acc, field) => {
            return {
              ...acc,
              [`slo.groupings.${field}`]: {
                terms: {
                  field,
                },
              },
            };
          },
          {
            'slo.instanceId': {
              terms: { field: 'slo.instanceId' },
            },
          }
        )
      : { 'slo.instanceId': { terms: { field: 'slo.instanceId' } } };
```

Reply:
There is something similar to be done in the this file as well: https://github.com/kdelemme/kibana/blob/main/x-pack/plugins/observability/server/services/slo/summary_transform_generator/generators/common.ts#L11